File: app/src/main/java/com/example/dulong/activity/user/OrderDetailActivity.java
package com.example.dulong.activity.user;

import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.dulong.R;
import com.example.dulong.activity.HomeActivity;
import com.example.dulong.activity.utils.CartManager;
import com.example.dulong.adapter.OrderAdapter;
import com.example.dulong.model.Product;

import java.text.DecimalFormat;
import java.util.List;
import java.util.Random;

public class OrderDetailActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_order_detail);

        // 1. Ánh xạ View
        Toolbar toolbar = findViewById(R.id.toolbarOrder);
        RecyclerView rvOrderProducts = findViewById(R.id.rvOrderProducts);
        TextView tvTotalPrice = findViewById(R.id.tvOrderTotalPrice);
        TextView tvOrderId = findViewById(R.id.tvOrderId);
        TextView btnCopyOrder = findViewById(R.id.btnCopyOrder);

        // Ánh xạ địa chỉ
        TextView tvNamePhone = findViewById(R.id.tvOrderNamePhone);
        TextView tvAddress = findViewById(R.id.tvOrderAddress);

        setSupportActionBar(toolbar);
        if (getSupportActionBar() != null) {
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
            getSupportActionBar().setDisplayShowTitleEnabled(false);
        }
        toolbar.setNavigationOnClickListener(v -> {
            Intent intent = new Intent(this, HomeActivity.class);
            startActivity(intent);
        });

        // 2. Nhận dữ liệu từ CheckoutActivity
        String namePhone = getIntent().getStringExtra("order_name_phone");
        if (namePhone == null) namePhone = "Không có tên";
        
        String address = getIntent().getStringExtra("order_address");
        if (address == null) address = "Không có địa chỉ";
        
        double total = getIntent().getDoubleExtra("order_total", 0.0);

        // Hiển thị địa chỉ
        tvNamePhone.setText(namePhone);
        tvAddress.setText(address);

        // 3. Hiển thị danh sách sản phẩm
        boolean orderSuccess = getIntent().getBooleanExtra("order_success", false);
        List<Product> orderedItems;
        
        if (orderSuccess) {
            // Lấy danh sách sản phẩm từ intent
            orderedItems = (List<Product>) getIntent().getSerializableExtra("ordered_items");
            if (orderedItems == null) {
                orderedItems = new java.util.ArrayList<>();
            }
            
            // Hiển thị thông báo thành công
            String orderMessage = getIntent().getStringExtra("order_message");
            if (orderMessage != null) {
                Toast.makeText(this, orderMessage, Toast.LENGTH_LONG).show();
            }
        } else {
            orderedItems = CartManager.getCartItems();
        }
        
        OrderAdapter adapter = new OrderAdapter(orderedItems);
        rvOrderProducts.setLayoutManager(new LinearLayoutManager(this));
        rvOrderProducts.setAdapter(adapter);

        // 4. Hiển thị tổng tiền
        DecimalFormat formatter = new DecimalFormat("#,###");
        tvTotalPrice.setText(formatter.format(total) + "đ");

        // 5. Tạo mã đơn hàng (Nếu chưa có ID từ Intent thì tạo mới)
        String randomId = getIntent().getStringExtra("order_id");
        if (randomId == null) {
            randomId = "251209" + (new Random().nextInt(900000) + 100000);
        }
        tvOrderId.setText(randomId);

        // Sự kiện sao chép mã
        final String finalRandomId = randomId;
        btnCopyOrder.setOnClickListener(v -> {
            ClipboardManager clipboard = (ClipboardManager) getSystemService(CLIPBOARD_SERVICE);
            ClipData clip = ClipData.newPlainText("Mã đơn hàng", finalRandomId);
            clipboard.setPrimaryClip(clip);
            Toast.makeText(this, "Đã sao chép mã đơn hàng", Toast.LENGTH_SHORT).show();
        });
    }
}

File: app/src/main/java/com/example/dulong/activity/user/OrderDetailActivity.kt
package com.example.dulong.activity.user

import android.content.ClipData
import android.content.ClipboardManager
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.dulong.R
import com.example.dulong.activity.utils.CartManager
import com.example.dulong.adapter.OrderAdapter // Bạn có thể dùng CartAdapter hoặc OrderAdapter nếu giống nhau
import java.text.DecimalFormat
import kotlin.random.Random

class OrderDetailActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_order_detail)

        // 1. Ánh xạ View
        val toolbar: Toolbar = findViewById(R.id.toolbarOrder)
        val rvOrderProducts: RecyclerView = findViewById(R.id.rvOrderProducts)
        val tvTotalPrice: TextView = findViewById(R.id.tvOrderTotalPrice)
        val tvOrderId: TextView = findViewById(R.id.tvOrderId)
        val btnCancel: Button = findViewById(R.id.btnCancelOrder)

        // View hiển thị địa chỉ (Thêm id vào layout activity_order_detail.xml nếu chưa có)
        // Ví dụ trong XML bạn cần đặt id cho TextView tên+sđt và TextView địa chỉ
        val tvNamePhone: TextView = findViewById(R.id.tvOrderNamePhone) // Cần thêm ID trong XML
        val tvAddress: TextView = findViewById(R.id.tvOrderAddress)     // Cần thêm ID trong XML

        setSupportActionBar(toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setDisplayShowTitleEnabled(false)
        toolbar.setNavigationOnClickListener { finish() }

        // 2. Nhận dữ liệu từ CheckoutActivity
        val namePhone = intent.getStringExtra("order_name_phone") ?: "Không có tên"
        val address = intent.getStringExtra("order_address") ?: "Không có địa chỉ"
        val total = intent.getDoubleExtra("order_total", 0.0)

        // Hiển thị địa chỉ
        tvNamePhone.text = namePhone
        tvAddress.text = address

        // 3. Hiển thị danh sách sản phẩm
        val orderedItems = CartManager.getCartItems()
        val adapter = OrderAdapter(orderedItems) // Hoặc dùng CartAdapter nếu OrderAdapter chưa có
        rvOrderProducts.layoutManager = LinearLayoutManager(this)
        rvOrderProducts.adapter = adapter

        // 4. Hiển thị tổng tiền
        val formatter = DecimalFormat("#,###")
        tvTotalPrice.text = "${formatter.format(total)}đ"

        // 5. Tạo mã đơn hàng
        val randomId = "251209${Random.Default.nextInt(100000, 999999)}"
        tvOrderId.text = randomId

        tvOrderId.setOnClickListener {
            val clipboard = getSystemService(CLIPBOARD_SERVICE) as ClipboardManager
            val clip = ClipData.newPlainText("Mã đơn hàng", randomId)
            clipboard.setPrimaryClip(clip)
            Toast.makeText(this, "Đã sao chép mã đơn hàng", Toast.LENGTH_SHORT).show()
        }

        // 6. Xử lý nút Hủy đơn hàng
        btnCancel.setOnClickListener {
            Toast.makeText(this, "Đã hủy đơn hàng thành công", Toast.LENGTH_SHORT).show()
            // Clear giỏ hàng nếu cần
            // CartManager.clearCart()
            finish()
        }
    }
}